bitcoin/pubkey.c

```diff
@@ -65,3 +65,8 @@ bool pubkey_eq(const struct pubkey *a, const struct pubkey *b)
 {
 	return structeq(&a->pubkey, &b->pubkey);
 }
+
+int pubkey_cmp(const struct pubkey *a, const struct pubkey *b)
+{
+	return memcmp(a, b, sizeof(*a));
```

Comment:
Yeah, I was hoping to avoid serialization, but relying on internal data layout is not a good idea.